msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,348 @@
->
-> * within your app's description section.
-> * in your app's welcome message.
-> * in your app's help message.
-> * in the window where you ask a user to sign-in to your services.
+### 3.3 Response time
 
-* Apps without an easy sign-up flow must also include a help tab or link to a web page, where a new user can see detailed guidance on configuring your Teams app. Provide detailed information to ensure a new user is not blocked when trying your app for the first time.
-* Sign-in and sign-out functionality must work on mobile clients. Ensure to use the [Microsoft Teams SDK](https://www.npmjs.com/package/@microsoft/teams-js) version 1.4.1 or later.
+Teams apps must respond within a reasonable timeframe, which varies depending on the capability.
 
-For additional information on authentication see:
+* Tabs must respond within three seconds or display a loading message or warning.
+* Bots must respond to user commands within two seconds or display a typing indicator.
+* Messaging extensions must respond to user commands within five seconds.
+* Notification must load within five seconds of the user action.
 
-* [Authentication documentation](../../../authentication/authentication.md)
-* [Bot authentication sample in Node](https://github.com/OfficeDev/microsoft-teams-sample-auth-node)
-* [Tab authentication sample in Node](https://github.com/OfficeDev/microsoft-teams-sample-complete-node)
-* [Tab/bot authentication in C#/.NET](https://github.com/OfficeDev/microsoft-teams-sample-complete-csharp)
+## 4.0 App package and store listing
 
-### &#9989; Response times must be reasonable
+App packages must be correctly formatted and include all required information and components.
 
-* **Tabs**. If a response to an action takes more than three seconds, you must provide a loading message or warning.
-* **Bots**. A response to a user command must occur within two seconds. If longer processing is required, your app must display a typing indicator.
-* **Compose extensions**. A response to a user command must occur within five seconds.
+### 4.1 App manifest
+
+* The app manifest must conform to the [latest manifest schema](~/resources/schema/manifest-schema.md).
+* If your app includes a bot or messaging extension, the manifest must be consistent with Bot Framework metadata, including bot name, logo, privacy policy link, and terms of service link.
+* If your app uses Azure Active Directory (Azure AD) for authentication, include the Azure AD ID in the manifest. For more information, see [manifest schema](~/resources/schema/manifest-schema.md#webapplicationinfo).
+
+### 4.2 App icons
+
+* Your app package must include two PNG versions of your app icon: a color icon and an outline icon.
+* Correctly sized and formatted icons must be specified in your app package. The icons must match what's submitted with the store listing metadata. For more information, see the Teams app [icon guidelines](~/concepts/build-and-test/apps-package.md#app-icons).
```

Comment:
@heath-hamilton Please update - 
Icons are one of the main elements users see when browsing the store. Your icons should communicate your app's brand purpose while also adhering to Teams requirements.
•	Your app package must include two PNG versions of your app icon: a color icon and an outline icon.
•	Correctly sized and formatted icons must be specified in your app package. The icons must match what's submitted with the store listing metadata. For more information, see the Teams app icon guidelines.
Color icon
The color version of your icon displays in most Teams scenarios and must be 192x192 pixels. Your icon symbol (96x96 pixels) can be any color or colors, but it must sit on a solid or fully transparent square background.
Teams automatically crops your icon to display a square with rounded corners in multiple scenarios and a hexagonal shape in bot scenarios. Include 48 pixels of padding around your symbol so these crops can be made without losing any detail.
 
 
Outline icon
An outline icon displays in two scenarios:
•	When your app is in use and “hoisted” on the app bar on the left of Teams.
•	when a user pins your app's messaging extension.
The icon must be 32x32 pixels. It can be white with a transparent background or transparent with a white background (no other colors are permitted). The outline icon should not have any extra padding around the symbol.